### ISSUE TITLE
As a CIF ops team member, I want to be able to enter milestone information, so that I can track their status and completion

### ISSUE THREAD

**Issue by pbastia** [bcgov/cas-cif#583]
#### Description:

Card about the form for milestones. General, Advance Payment, etc... could all be aggregated in one, or separated on the interface?
Maybe for the first iteration, we can make them all the same, and reassess on the next round of ux studies @suhafa  @nanyangpro 

#### Acceptance Criteria:

Given
When
Then

#### Development Checklist:

- [ ] Checklist item
- [ ] Checklist item
- [ ] Checklist item
- [ ] Meets the DOD

**Definition of Ready** (Note: If any of these points are not applicable, mark N/A)

- [ ] User story is included
- [ ] User role and type are identified
- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if required)
- [ ] Design / Solution is accepted by Product Owner
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

·**Definition of Done** (Note: If any of these points are not applicable, mark N/A)

- [ ] Acceptance criteria are tested by the CI pipeline
- [ ] UI meets accessibility requirements
- [ ] Configuration changes are documented, documentation and designs are updated
- [ ] Passes code peer-review
- [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
- [ ] Ticket is ready to be merged to main branch
- [ ] Can be demoed in Sprint Review
- [ ] Bugs or future work cards are identified and created
- [ ] Reviewed and approved by Product Owner

#### Notes:

-

**Comment by suhafa** [comment]
Good idea to keep the first iteration aa simple form, the interface doesn't differ much amongst milestone types. We can add plug the dependencies in subsequent iterations. Listing them here for future reference – 

Different Milestone types would trigger a few changes on the Milestone report form. For instance:
- For a '**Reporting**' milestone type, there is no funding associated with it, so Maximum Funding Amount can be set to 'Not Applicable' by default.
- The time period for substantial completion date differs depending on the milestone type. It is 30 days for general milestones, 60 days for project summary report (IA - final milestone) and 150 days for TEIMP report (EP Final Milestone). We can set the substantial completion date automatically depending on the milestone type chosen.
- IA Funding stream typically has only **Advanced Milestone + Project Summary Report** Milestone only. There may be Reporting moving forward. Perhaps we can automate this funding stream type to have two kinds of milestones created by default once the funding stream is selected.

**Comment by pbastia** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Climate-Action-Secretariat-60ca4121764d710011481ca2/issues/bcgov/cas-cif/583?planning-poker) with ZenHub @BCerki @dleard @gurjmatharu @matthieu-foucault @mikevespi @Sepehr-Sobhani

**Comment by elliottSlee** [comment]
@pbastia I see the development checklist includes "Editing and Deleting milestones form changes using existing mutations". I was going to create a separate ticket for deleting milestones, is that necessary?

**Comment by pbastia** [comment]
@elliottSlee Not really. We're reusing components and patterns built for all reporting requirements, no new feature is needed for that

**Comment by LindsayMacfarlane** [comment]
Thank you @dleard ! I tested in dev by adding, editing and deleting milestones.

A couple of observations during testing:

- When I went to edit an existing project, there was no 'edit button' for the milestone report like there are for the other forms. Not sure if this was intentional? So, as a user, if I clicked the milestone in the task list, it was automatically editable. 

- time out issues: I was logged out every 2 minutes and the 'remain active' button did not work (i.e., it wouldn't keep me logged in). I believe this will be covered in ticket #429)

**Comment by dleard** [comment]
Good observations, there is a card in the sprint backlog that will create a 'read-only' version of milestones to be in line with the other pages (not editable right away & show an edit button) #587
Yes, I think the timeout issue will be covered in #429
